components/NavigationDrawer/src/MDCBottomDrawerHeader.h

```diff
@@ -0,0 +1,32 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ A protocol view controllers should conform to for enabling the view controller to be
```

Comment:
This can't be implemented by a UIView object? It looks like in the examples there's no work to be done, so it seems like a VC isn't strictly required.

Reply:
This is not recommended , but I will revert to @guylivneh on this to respond after the weekend as he knows more.

Reply:
Hi, chiming in - first, we like to be consistent with when we use views and view controllers - we do here the same for the header as we do for the main content. Secondly, the header is expected to change its state while transitioning to fullscreen, so we expect some logic to be involved. Taking in a view controller has the benefit for supporting a header with more logic. 

Saying that, I'll also want to lay out the use case of how a header view controller (fooHeaderViewController) could be used with the drawer if we choose to take in a UIView instead of a UIViewController as the drawer's header: in that case, a user will have to complete two steps to add its header to the drawer:
1) Set the drawer's headerView property to fooHeaderViewController.view.
2) Add fooHeaderViewController as a child view controller of the drawer view controller instance.
This can be done but the cost is overcomplicating the usage case and making the use more error prone.

Wdyt?

Reply:
thanks for the input @guylivneh!